EIPS/eip-2477.md

```diff
@@ -0,0 +1,211 @@
+
+The digest return value is first, this is an optimization because we expect on-chain implementations will be more likely to use this return value if they will only be using one of the two.
+
+**Function return types**
+
+The digest is a byte array and supports various hash lengths. This is consistent with SRI. Whereas SRI uses base64 encoded to target a HTML document, we use a byte array because Ethereum already allows this encoding.
+
+:warning: TODO: WE NEED TO SPECIFY ENDIANNESS ABOVE AND PROVIDE TEST CASES BELOW. AND JUSTIFY THAT HERE.
+
+The hash function name is a string. Currently there is no universal taxonomy of hash function names. SRI recognizes the names `sha256`, `sha384` and `sha512` with case-insensitive matching. We are aware of two authorities which provide taxonimies and canonical names for hash functions: ETSI Object Identifiers and NIST Computer Security Objects Register. However, SRI's approach is easier to follow and we have adopted this here.
+
+**Function return type — hash length**
+
+Clients must support the SHA-256 algorithm may support others. This is a departure from the SRI specification here SHA-256, SHA-384 and SHA-512 are all required. The rationale for this less-secure requirement is because we expect some clients to be on-chain. Currently SHA-256 is simple and cheap to do on Ethereum whereas SHA-384 and SHA-512 are more expensive and cumbersome.
+
+The most popular hash function size below 256 bits in current use is SHA-1 at 160 bits. Multiple collissions (the "Shattered" PDF file, the 320 byte file, the chosen prefix) have been published and a recipe is given to generate infinitely more collisions. SHA-1 is broken. The United States National Institute of Standards and Technology (NIST) has first deprecated SHA-1 for certain use cases in November 2015 and has later further expanded this deprecation.
+
+The most popular hash function size above 256 bits in current use is SHA-384 as specified by NIST.
+
+The United States National Security Agency requires a hash length of 384 or more bits for the SHA-2 (CNSA Suite Factsheet) algorithm suite for use on TOP SECRET networks. (No unclassified documents are currently available to specify use cases at higher classification networks.)
+
+We suspect that SHA-256 and the 0xcert Asset Certification will be popular choices to secure token metadata for the foreseeable future.
+
+## Backwards Compatibility
+
+Both ERC-721 and ERC-1155 provide compatible token metadata specifications built on top of URIs and using JSON schemas. This standard is compatible with both and all specifications are addititive. Therefore, there are no backwards compatability regressions.
+
+ERC-1523 Standard for Insurance Policies as ERC-721 Non Fungible Tokens (DRAFT) proposes an extension to ERC-721 which also tightens the requirements on metadata. Because this is wholly an extension of ERC-721 and this standard ERC-2477 already supports ERC-721 then therefore ERC-1523 is automatically supported.
```

Comment:
Suggest rewrite to:
ERC-1523 Standard for Insurance Policies as ERC-721 Non Fungible Tokens (DRAFT) proposes an extension to ERC-721 which also tightens the requirements on metadata. Because it is wholly an extension of ERC-721, ERC-1523 is automatically supported by ERC-2477 (since this standard already supports ERC-721).